### PR TITLE
Drop deprecated code from Bundles & Bridges

### PR DESCRIPTION
**Pull request by stloyd**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | yes |
| Deprecations? | no |
| Tests pass? | yes |
| License | MIT |

### CODE REVIEW

File: src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php [hunk 1]
```diff
@@ -16,6 +16,9 @@
 
 class ContainerAwareEventManagerTest extends \PHPUnit_Framework_TestCase
 {
+    protected $container;
+    protected $evm;
```
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php]: use private vibility please (and this should be submitted separately as a bugfix as it should not go only in master)

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php [hunk 2]
```diff
@@ -706,7 +706,6 @@ private function registerTranslatorConfiguration(array $config, ContainerBuilder
             ;
 
             foreach ($finder as $file) {
-                list($domain, $locale, $format) = explode('.', $file->getBasename(), 3);
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: are you sure about this ? It is not removing a deprecated code

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/ActionsHelper.php [hunk 3]
```diff
@@ -58,9 +58,7 @@ public function controller($controller, $attributes = array(), $query = array())
     }
 
     /**
-     * Returns the canonical name of this helper.
-     *
-     * @return string The canonical name
+     * {@inheritdoc}
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/ActionsHelper.php]: this should be submitted to older branches actually

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/RequestHelper.php [hunk 4]
```diff
@@ -69,26 +60,24 @@ public function getLocale()
...
+    public function getName()
+    {
+        return 'request';
+    }
+
     private function getRequest()
     {
-        if ($this->requestStack) {
-            if (!$this->requestStack->getCurrentRequest()) {
+        if (null === $this->requestStack) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/RequestHelper.php]: it cannot be null anymore (and your change is wrong btw, as you are changing the meaning of the condition, making broken)

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/RequestHelper.php [hunk 5]
```diff
@@ -28,20 +28,11 @@ class RequestHelper extends Helper
...
-     *
-     * @deprecated since version 2.5, passing a Request instance is deprecated and support for it will be removed in 3.0.
+     * @param RequestStack $requestStack A RequestStack instance
      */
-    public function __construct($requestStack)
+    public function __construct(RequestStack $requestStack)
     {
-        if ($requestStack instanceof Request) {
-            trigger_error('Since version 2.5, passing a Request instance into the '.__METHOD__.' is deprecated and support for it will be removed in 3.0. Inject a Symfony\Component\HttpFoundation\RequestStack instance instead.', E_USER_DEPRECATED);
-            $this->request = $requestStack;
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/RequestHelper.php]: ``$this->request`` should now be removed entirely

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/SessionHelper.php [hunk 6]
```diff
@@ -28,20 +27,11 @@ class SessionHelper extends Helper
...
-     *
-     * @deprecated since version 2.5, passing a Request instance is deprecated and support for it will be removed in 3.0.
+     * @param RequestStack $requestStack A RequestStack instance
      */
-    public function __construct($requestStack)
+    public function __construct(RequestStack $requestStack)
     {
-        if ($requestStack instanceof Request) {
-            trigger_error('Since version 2.5, passing a Request instance into the '.__METHOD__.' is deprecated and support for it will be removed in 3.0. Inject a Symfony\Component\HttpFoundation\RequestStack instance instead.', E_USER_DEPRECATED);
-            $this->session = $requestStack->getSession();
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/SessionHelper.php]: ``$this->session`` should be removed as it is not set anymore at all

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/SessionHelper.php [hunk 7]
```diff
@@ -72,26 +62,24 @@ public function hasFlash($name)
...
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'session';
+    }
+
     private function getSession()
     {
         if (null === $this->session) {
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/SessionHelper.php]: it is always null now

File: src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php (hunk 1 above)
Comment (stloyd) [inline src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php]: It was already done #14128.

File: src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php (hunk 2 above)
Comment (stloyd) [inline src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php]: Yes, this code is not used in 2.7 anymore, I will backport it...

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php [hunk 8]
```diff
@@ -399,32 +383,6 @@ public function testFormsCanBeEnabledWithoutCsrfProtection()
...
     }
 
-    /**
-     * @group legacy
-     */
-    public function testLegacyFormCsrfFieldNameCanBeSetUnderCsrfSettings()
-    {
-        $this->iniSet('error_reporting', -1 & ~E_USER_DEPRECATED);
-
-        $container = $this->createContainerFromFile('form_csrf_sets_field_name');
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: you should remove these fixture files too

File: src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php [hunk 9]
```diff
@@ -25,33 +23,15 @@
 class LogoutUrlHelper extends Helper
 {
     private $generator;
-    private $listeners = array();
-    private $router;
-    private $tokenStorage;
```
Comment (stof) [inline src/Symfony/Bundle/SecurityBundle/Templating/Helper/LogoutUrlHelper.php]: if these are already unused in the current code, please submit a PR removing them in the 2.7 branch to clean the code

File: src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php (hunk 1 above)
Comment (stof) [inline src/Symfony/Bridge/Doctrine/Tests/ContainerAwareEventManagerTest.php]: then it should not appear in this diff

File: src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php (hunk 8 above)
Comment (stloyd) [inline src/Symfony/Bundle/FrameworkBundle/Tests/DependencyInjection/FrameworkExtensionTest.php]: Done.

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/RequestHelper.php [hunk 10]
```diff
@@ -69,26 +58,20 @@ public function getLocale()
...
+     * {@inheritdoc}
      */
     public function getName()
     {
         return 'request';
     }
+
+    private function getRequest()
+    {
+        if ($this->requestStack && $currentRequest = $this->requestStack->getCurrentRequest()) {
```
Comment (sstok) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/RequestHelper.php]: The requestStack property can't be null so no need to check if it is set.

**Comment by stof** [comment]
@fabpot @nicolas-grekas how are we dealing with these removal PRs which are starting to come ? Are we letting @stloyd make a bug PR here, or are we dealing with many smaller PRs ?

**Comment by fabpot** [comment]
Less PRs about this topic, the better. If we can centralize everything in one PR, that would be even better.

**Comment by nicolas-grekas** [comment]
The hardest part will be keeping tests green on both 3.0 and 2.7 by not forgetting to update composer dependencies. See an example of this in #14216

**Comment by dosten** [comment]
Can you rebase?

File: src/Symfony/Bundle/FrameworkBundle/Templating/Helper/SessionHelper.php [hunk 11]
```diff
@@ -72,26 +61,20 @@ public function hasFlash($name)
         return $this->getSession()->getFlashBag()->has($name);
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Templating/Helper/SessionHelper.php]: Request::getSession says `@return SessionInterface|null` so this is not save to be called.

File: src/Symfony/Bridge/Twig/AppVariable.php [hunk 12]
```diff
@@ -14,7 +14,6 @@
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\RequestStack;
 use Symfony\Component\HttpFoundation\Session\Session;
-use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
```
Comment (GromNaN) [inline src/Symfony/Bridge/Twig/AppVariable.php]: This is used in a comment.
https://github.com/stloyd/symfony/blob/feature/drop_depr/src/Symfony/Bridge/Twig/AppVariable.php#L56

**Comment by fabpot** [comment]
Closing this old PR which is now obsolete and does too many things.

**Comment by dosten** [comment]
@fabpot What is the way to go with the removal of deprecated features? Small PRs for each bundle/component/bridge? I'm asking because this is the oposite of what did you say in a [previous comment](https://github.com/symfony/symfony/pull/14127#issuecomment-89281600)

**Comment by fabpot** [comment]
@dosten One PR would have been great but it must be ready to be merged. As this one is quite old and because we merged so many removal since then, it would make rebasing this PR difficult. The best now is indeed to do small PRs for one component or one bundle.

**Pull request closed without merging**